src/drivers/intel/hda/hda-dma.c

```diff
@@ -348,9 +348,8 @@ static int hda_dma_host_start(struct dma_chan_data *channel)
 	struct hda_chan_data *hda_chan = dma_chan_get_data(channel);
 	int ret = 0;
 
-	/* Force Host DMA to exit L1 only on start*/
-	if (!(hda_chan->state & HDA_STATE_RELEASE))
-		pm_runtime_put(PM_RUNTIME_HOST_DMA_L1, 0);
+	/* Force Host DMA to exit L1 on start */
+	pm_runtime_put(PM_RUNTIME_HOST_DMA_L1, 0);
```

Comment:
I think the original logic is that we don't need to force L1 exit for the pause/release iteration, while we need it for a _start, as there will be preloading for a hda_dma_start() while there isn't for the _release().

We do have dma_pause() and dma_release() calling in dai.c, even though there might not be trigger pause/release IPCs from the Linux host side, so I would suggest to keep this as it is in case it doesn't bring any issues like Xruns to us.

Reply:
@keyonjie DAI uses link DMA and we have the ops for link DMA here. There is no need for host DMA pause/release when it is not used